ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -440,30 +447,33 @@ impl<'a> Assemble<'a> {
                 }
                 // Store the future patch offset for this guard.
                 let off = self.asm.offset();
-                self.deoptinfo.get_mut(&deoptid).unwrap().fail_offset = off;
+                self.deoptinfo.get_mut(&deoptid).unwrap().1.fail_offset = off;
                 // Emit the guard failure code.
                 let deoptid = i32::try_from(deoptid).unwrap();
                 dynasm!(self.asm
                     ;=> fail_label
-                    // After compiling a side-trace for this location, we want to patch its address
-                    // here, so we can jump to side-traces directly. The jump instruction requires
-                    // 5 bytes, but since we have to patch this with 8 bytes, we leave 8 bytes
-                    // worth of nops (we could leave less nops if we were to overwrite the
+                    // After compiling a side-trace for this location, we want to patch in a jump
+                    // to its address here later, so we can jump to side-traces directly. Depending
+                    // on the distance of the side-trace to the parent trace we can either emit a
+                    // 32-bit jump (5 bytes) or need to patch a more involved 64-bit jump (22
+                    // bytes). We will leave enough space for either. The jump is required so that
+                    // we can patch the space following it with instructions to do the 64-bit jump
+                    // without having other threads pick up partial changes. Only in the last
+                    // atomic operation is the jump removed and we fall through to the 64-bit jump.
+                    // FIXME: we currently leave 36 bytes of patch space, though we likely need
```

Comment:
36 seems a weird number? Shouldn't we leave at most `x - 1` (where `x` is a multiple of 8) bytes?

Reply:
You are right and I'm bad at maths. It's supposed to be 32.

Reply:
This is no longer how we do things.